l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -477,4 +490,61 @@ contract ExecutorFacet is Base, IExecutor {
     function _setBit(uint256 _bitMap, uint8 _index) internal pure returns (uint256) {
         return _bitMap | (1 << _index);
     }
+
+    /// @notice Calls the point evaluation precompile and verifies the output
+    // Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof.
+    // Also verify that the provided commitment matches the provided versioned_hash.
+    function _pointEvaluationPrecompile(bytes32 _versionedHash, bytes32 _inputPoint, bytes calldata _valueCommitmentProof) internal returns (bytes32) {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _inputPoint, _valueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.call(precompileInput);
+
+        // ToDo: Check output against spec, should have field elements prepended but other clients dont use it.
+        require(success, "failed to call point evaluation precompile");
+        uint256 result = abi.decode(data, (uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    function _verifyBlobInformation(bytes calldata _pubdataCommitments, bytes32 _blob1Hash, bytes32 _blob2Hash) internal returns (bytes32 blobCommitmentRollingHash) {
+        uint256 versionedHashIndex = 0;
+
+        // ToDo: This should be dynamic instead of being hardcoded to 2 blobs
+        require(_pubdataCommitments.length == PUBDATA_COMMITMENT_SIZE * 2, "bs");
```

Comment:
are you sure the constant is correct here? the `i` in the loop below is incrememented at `POINT_EVALUATION_INPUT_SIZE` at a time, so intuitively we would want `POINT_EVALUATION_INPUT_SIZE * 2`, but maybe I am missing something

Reply:
good catch, we actually want the loop to be incremented by `PUBDATA_COMMITMENT_SIZE` since we derive the other parts (versioned hash, opening point) of the `POINT_EVALUATION_INPUT_SIZE` from other sources

Reply:
but this also depends on if we are pulling the point to perform the evaluation from calldata or recomputing it on chain